rust/ql/lib/codeql/rust/security/regex/RegexInjectionQuery.qll

```diff
@@ -0,0 +1,31 @@
+/**
+ * Provides a taint-tracking configuration for detecting regular expression
+ * injection vulnerabilities.
+ */
+
+private import rust
+private import codeql.rust.dataflow.DataFlow
+private import codeql.rust.dataflow.TaintTracking
+private import codeql.rust.dataflow.FlowSource
+private import codeql.rust.Concepts
+private import codeql.rust.security.regex.RegexInjectionExtensions
+
+/**
+ * A taint configuration for detecting regular expression injection vulnerabilities.
+ */
+module RegexInjectionConfig implements DataFlow::ConfigSig {
+  predicate isSource(DataFlow::Node source) { source instanceof ThreatModelSource }
```

Comment:
I've not fully looked into this, but I would've though this should have been `ActiveThreatModelSource`, but that made the test results disappear.

Reply:
It should indeed be `ActiveThreatModelSource`, but then your tests will have to use a remote rather than a local source.  OR we could turn on local sources as active threat models for tests.  I've thought about doing the latter in the past and I do have instructions for how to do it somewhere.

Reply:
I used `std::env::var` because it seemed like the easiest way to get some tainted input. Is there a similarly easy remote source? If so I could use that, otherwise I think activating local sources seems preferable.

Reply:
> Is there a similarly easy remote source?

Not really, the simplest is `reqwest` but that's still a few extra steps.  I think we should enable local sources as active threat models in tests, both to support simpler tests like the ones you've written, and to make sure we see all potential results in the tests regardless of user configuration.

I'll DM you the instructions I was given for doing this...

Reply:
Thanks for addressing this.